core/src/test/java/org/elasticsearch/common/io/stream/BytesStreamsTests.java

```diff
@@ -431,6 +431,29 @@ public String getWriteableName() {
                 endsWith(" claims to have a different name [intentionally-broken] than it was read from [test-named-writeable]."));
     }
 
+    public void testWriteStreamableList() throws IOException {
+        final int size = randomIntBetween(0, 5);
+        final List<TestStreamable> expected = new ArrayList<>(size);
+
+        for (int i = 0; i < size; ++i) {
+            expected.add(new TestStreamable(randomBoolean()));
+        }
+
+        final BytesStreamOutput out = new BytesStreamOutput();
+
+        out.writeStreamableList(expected);
+
+        final StreamInput in = StreamInput.wrap(out.bytes().toBytes());
+
+        List<TestStreamable> loaded = in.readStreamableList(TestStreamable::new);
+
+        assertThat(loaded, hasSize(expected.size()));
+
+        for (int i = 0; i < expected.size(); ++i) {
+            assertEquals(expected.get(i).value, loaded.get(i).value);
+        }
+    }
```

Comment:
You can also assert that there are no bytes left to read I believe. Some of the other tests in this file do it and it is a good idea I think.